# Analyzing of Mobile Apps Data

Hello, this is my first project in Jupyter and the aim of this project is to find mobile apps which are profitable in Apple store and Google Play Market.
All sample data sets are available online:

1. **[Google](https://www.kaggle.com/lava18/google-play-store-apps)** 10k data set.
2. **[Apple](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)** 7k data set.

In these datasets I m going to analyze apps that are free to downlowd and install. These apps are considered as ones that are having main source of revenue from in-app adds.

There are many apps in different kind of languages. This project will analyze only apps in English.


## Opening and exploring data

In March 2021 there are 1,96 milion apps in the Apple App Store and 2,87 milion apps in the Google Play Store. To manage such big amount of datas takes a time, so within this project I m going to check only samples of data mentioned above.

In [3]:
from csv import reader

# Apple apps dataset opening #
open_file = open('AppleStore.csv')
read_file = reader(open_file)
apple = list(read_file)
apple_header = apple[0]
apple = apple[1:]

# Android apps dataset opening #
open_file = open('googleplaystore.csv')
read_file = reader(open_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

print(apple_header)
print('\n')
print(android_header)
print('\n')
print(apple[:5])
print('\n')
print(android[:5])
print('\n')

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


[['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1'], ['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1'], ['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1'], ['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1'], ['284035177', 'Pandora - Music & Radio', '130242560', 'U

With the explore() function I am trying to understand few rows of the dataset. 

In [4]:
def explore_data(dataset, start, end, rows_and_columns=False):
    data_slice = dataset[start:end]
    for row in data_slice:
        print(row)
        print('\n')
    
    if rows_and_columns:
        print('Number of rows', len(dataset))
        print('Number of columns', len(dataset[0]))

        
#app store #
print(apple_header)
print('\n')
explore_data(apple, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows 7197
Number of columns 16


In [5]:
# google play store #

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows 10841
Number of columns 13


### Finding and deleting wrong data 

In [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion) under *Google Play Store data set* was mentioned row **10472** as the row where one entry is missing. 

Each row should normaly have 13 entries but this particular row has only 12 entries.

To eliminate entry of this row *del* statement is used.

In [6]:
for row in android:
    headerlength = len(android_header)
    rowlength = len(row)
    
    if rowlength != headerlength:
        print(row)
        print('\n')
        print(android.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


10472


In [7]:
print(android[10472]) #wrong row
print('\n')
print(android[0])   #correct row
print('\n')
print(android_header)

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [8]:
print(len(android)) #nr of rows before using del statement
del android[10472]  #run this row just once !!! 10840 after executed row
print(len(android)) #nr of rows after using del statement

10841
10840


### Removing duplicate entries ###

In *Google Apps* dataset there is a chance that some apps have more than one entries. Let´s check if it´s true and find number of unique and number of duplicate app entries.

In [9]:
duplicate = []
unique = []

for row in android:
    if row[0] in unique:
        duplicate.append(row[0])
    else:
        unique.append(row[0])
        
print('Number of duplicate apps', len(duplicate))
print('Number of unique apps', len(unique))
print('\n')
print('Example of duplicate apps', duplicate[:15])

Number of duplicate apps 1181
Number of unique apps 9659


Example of duplicate apps ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


From result above we can see that there are aprox. 10% of duplicates apps. Those duplicates will be removed based on *'Reviews'* value. I will keep the entry with the latest (highest) review value and other entries will be removed. 

In [10]:
for row in android:
    if row[0] == "Google Ads":
        print(row)
        print('\n')

['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29313', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']


['Google Ads', 'BUSINESS', '4.3', '29331', '20M', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 30, 2018', '1.12.0', '4.0.3 and up']




Above you can see an example of duplicate entry. Here we have **Google Ads** app that has 3 entries. Entry with *29331* reviews  is the one I´ll keep. 

Below is created a dictionary with highest number of review of each unique app name.

In [11]:
reviews_max = {}

for row in android:
    name = row[0]
    n_reviews = float(row[3])
    
    #checking review value for each entry and appending those one
    #with the highest review to a dictionary
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    
    #new entries are appended to dictionary
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
    
print('Number of apps in review_max: ', len(reviews_max))

Number of apps in review_max:  9659


New dictionary was created above with unique apps as the key with it´s max review which help delete duplicate apps.

In [12]:
android_clean = [] #new list for cleaned data
already_added = [] #for storing app names

for row in android:
    name = row[0]
    n_reviews = float(row[3])
    
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(row)
        already_added.append(row[0])
        
print(len(android_clean))
    

9659


Using the max review dictionary we compare the values with the actual dataset 'review' values and append the unique matches into the `android_clean[]` list of lists. In order to avoid re addition of any app, I keep track of cleaned app names through `already_added[]`.

Now is time to check duplicates in **Appstore Dataset**

In [13]:
duplicate = []
unique = []

for row in apple:
    if row[0] in unique:
        duplicate.append(row[0])
    else:
        unique.append(row[0])
        
print('Number of duplicate apps', len(duplicate))
print('Number of unique apps', len(unique))
print('\n')
print('Example of duplicate apps', duplicate[:15])

Number of duplicate apps 0
Number of unique apps 7197


Example of duplicate apps []


In Appstore Dataset did not appear any duplicates.

### Removing Non-English Apps
 
This analyze will work only with apps for English-speaking users. It means that app names should not have any other characters than english. 

For this purpose will be used `ord()` built-in function to find the unicode values of characters. Any character with codes greater than 127 can be eliminated since in ASCII coding all english characters are contained in the range *0 - 127*.

Let´s try it first on few app names below.

In [14]:
def english(string):
    for element in string:
        if ord(element) > 127:
            return False
    
    return True

#function output is checked for various inputs
app1 = english('Instagram')
app2 = english('爱奇艺PPS -《欢乐颂2》电视剧热播')
app3 = english('Docs To Go™ Free Office Suite')
app4 = english('Instachat 😜')

print(app1, app2, app3, app4)

True False False False


Apps Docs To Go™ Free Office Suite and Instachat 😜 are english apps but they contain special characters like ™ and 😜 in their names, which are out of ASCII 0 - 127 range, so the names might be eliminated from the list.

To minimize the impact of data loss, I remove an app if its name has more than 3 characters with corresponding numbers falling outside the ASCII range. So this means all English apps with up to 3 emoji or other special characters will remain in the list.

In [15]:
def only_english(a_string):
    ch = 0
    for element in a_string:
        if ord(element) > 127:
            ch += 1
            if ch > 3:
                return False
    return True

app5 = only_english('Instagram')
app6 = only_english('爱奇艺PPS -《欢乐颂2》电视剧热播')
app7 = only_english('Docs To Go™ Free Office Suite')
app8 = only_english('Instachat 😜')

print(app5, app6, app7, app8)

True False True True


Result above showed that function `only_english` is more powerfull than `english` so that function can be used for the removal of Non English Apps.

In [16]:
def remove_NonEnglish(dataset, name_index):
    dataset_english = []  #list of english apps
    
    for row in dataset:
        name = row[name_index]
        if only_english(name) == True:
            dataset_english.append(row)
            
    print('Number of rows before removal: ', len(dataset))
    print('Number of rows after removal: ', len(dataset_english))
    print('Number of removed rows: ', len(dataset) - len(dataset_english))
    print('\n')
    
    return dataset_english

#calling function for both datasets
apple_english = remove_NonEnglish(apple, 1)
android_english = remove_NonEnglish(android_clean, 0)


Number of rows before removal:  7197
Number of rows after removal:  6183
Number of removed rows:  1014


Number of rows before removal:  9659
Number of rows after removal:  9614
Number of removed rows:  45




In last step there were succesfully removed **1014** non english apps from Apple dataset and **45** non english apps from Android dataset.

### Isolating apps which are free to download

In this project we are interested only in free apps, so now I go through `price` column and isolate apps which are free to download.

In [17]:
final_apple = []
final_android = []

for app in apple_english:
    price = app[4]
    if price == '0.0' or price == '$0':
        final_apple.append(app)

for app in android_english:
    price = app[7]
    if price == '0' or price == '$0':
        final_android.append(app)
        
print('Free apps in Apple dataset: ', len(final_apple))
print('Free apps in Android dataset: ', len(final_android))


Free apps in Apple dataset:  3222
Free apps in Android dataset:  8864


### Most common App by Genre

In this step I need to build a frequency table for the `prime_genre`column of the App Store data set and for the `Genres` and `Category`columns of the Google Play data set.
 
Two function will be built to analyze frequency tables:
1. First function will generate frequency tables that show percentage
2. Second function will display the percentage with descending order


In [21]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    #Appending values to table{} to create required frequency table
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    #Converting frequencies to percentage table
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index) #Calling for Percentage table
    table_display = []
    
    #Interchanging the values and keys position in list of tuples for easy sorting
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
    
    #Display table in descending order using sort      
    table_sorted = sorted(table_display, reverse=True)
    
    for entry in table_sorted:
        #Displaying values as % upto 2 decimals using round()
        print(entry[1], ':', round(entry[0],2), '%')


Checking frequency table for the *prime_gender* collumn of the App data set

In [25]:
display_table(final_apple, -5) 

Games : 58.16 %
Entertainment : 7.88 %
Photo & Video : 4.97 %
Education : 3.66 %
Social Networking : 3.29 %
Shopping : 2.61 %
Utilities : 2.51 %
Sports : 2.14 %
Music : 2.05 %
Health & Fitness : 2.02 %
Productivity : 1.74 %
Lifestyle : 1.58 %
News : 1.33 %
Travel : 1.24 %
Finance : 1.12 %
Weather : 0.87 %
Food & Drink : 0.81 %
Reference : 0.56 %
Business : 0.53 %
Book : 0.43 %
Navigation : 0.19 %
Medical : 0.19 %
Catalogs : 0.12 %


Result above showed, that among free apps in the App store, more than a half (58,16 %) are games. Second place is for entertainment with almost (8 %) and third is photo & video. Education is on the forth place with only 3,66 %.

Let's now check the genres and category columns of the Google Play data set (because those two columns might be related).

In [26]:
display_table(final_android, 1) #category

FAMILY : 18.91 %
GAME : 9.72 %
TOOLS : 8.46 %
BUSINESS : 4.59 %
LIFESTYLE : 3.9 %
PRODUCTIVITY : 3.89 %
FINANCE : 3.7 %
MEDICAL : 3.53 %
SPORTS : 3.4 %
PERSONALIZATION : 3.32 %
COMMUNICATION : 3.24 %
HEALTH_AND_FITNESS : 3.08 %
PHOTOGRAPHY : 2.94 %
NEWS_AND_MAGAZINES : 2.8 %
SOCIAL : 2.66 %
TRAVEL_AND_LOCAL : 2.34 %
SHOPPING : 2.25 %
BOOKS_AND_REFERENCE : 2.14 %
DATING : 1.86 %
VIDEO_PLAYERS : 1.79 %
MAPS_AND_NAVIGATION : 1.4 %
FOOD_AND_DRINK : 1.24 %
EDUCATION : 1.16 %
ENTERTAINMENT : 0.96 %
LIBRARIES_AND_DEMO : 0.94 %
AUTO_AND_VEHICLES : 0.93 %
HOUSE_AND_HOME : 0.82 %
WEATHER : 0.8 %
EVENTS : 0.71 %
PARENTING : 0.65 %
ART_AND_DESIGN : 0.64 %
COMICS : 0.62 %
BEAUTY : 0.6 %


On the first sight it looks much more optimistic than it was before in App Store. On the first place we have apps designed for Family (almost 19%) when Games are 2nd with (9,72 %). Further positions are for practical app categories like Tools, Business, Lifestyle. 

But when you open the Google Play store and get to the Family category you can see that aplications there are mostly games for kids. 

In [27]:
display_table(final_android, -4)

Tools : 8.45 %
Entertainment : 6.07 %
Education : 5.35 %
Business : 4.59 %
Productivity : 3.89 %
Lifestyle : 3.89 %
Finance : 3.7 %
Medical : 3.53 %
Sports : 3.46 %
Personalization : 3.32 %
Communication : 3.24 %
Action : 3.1 %
Health & Fitness : 3.08 %
Photography : 2.94 %
News & Magazines : 2.8 %
Social : 2.66 %
Travel & Local : 2.32 %
Shopping : 2.25 %
Books & Reference : 2.14 %
Simulation : 2.04 %
Dating : 1.86 %
Arcade : 1.85 %
Video Players & Editors : 1.77 %
Casual : 1.76 %
Maps & Navigation : 1.4 %
Food & Drink : 1.24 %
Puzzle : 1.13 %
Racing : 0.99 %
Role Playing : 0.94 %
Libraries & Demo : 0.94 %
Auto & Vehicles : 0.93 %
Strategy : 0.91 %
House & Home : 0.82 %
Weather : 0.8 %
Events : 0.71 %
Adventure : 0.68 %
Comics : 0.61 %
Beauty : 0.6 %
Art & Design : 0.6 %
Parenting : 0.5 %
Card : 0.45 %
Casino : 0.43 %
Trivia : 0.42 %
Educational;Education : 0.39 %
Board : 0.38 %
Educational : 0.37 %
Education;Education : 0.34 %
Word : 0.26 %
Casual;Pretend Play : 0.24 %
Music : 0.2 %
R

There is not very clear the difference between *Genders* and *Category* columns, but one thing is noticeable, that *Genders* are having much more categories. 


### Popular Apps by Genre in the App Store
When we want to check what apps are the most popular by genres, or which are having the most users, we needs to calculate the avg nr of installs for each app genre. This information is missing in App Store data set so that´s why there is need to calculate it. 
In Google Play data set this information is in *Installs* column.

In [32]:
genres_ios = freq_table(final_apple, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in final_apple:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', round(avg_n_ratings, 2))

Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22788.67
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


The result showed that navigation apps have the highest number of user reviews, but this might be a lot influenced by Waze and Google Maps applications which are having together almost half a milion user reviews, as you can see below..

In [33]:
for app in final_apple:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


The same situation is for social networking apps, where the average number is influenced a lot by giants like Facebook, Pinteres, Skype... Same is for music, where are again another giants like Spotify, Shazam or Pandora who influenced numbers.

In [34]:
for app in final_apple:
  if app[-5] == 'Social Networking':
    print(app[1], ':', app[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [35]:
for app in final_apple:
  if app[-5] == 'Music':
    print(app[1], ':', app[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

Reference apps are having 74,942 user ratings on average, which is also pretty hight. These numbers are skew up by the Bible app and Dictionary.com:

In [36]:
for app in final_apple:
  if app[-5] == 'Reference':
    print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


The main aim was to find popular genres amongs applications. But some genres might have higher average rates and by this they seems more popular than they really are. Those apps have higher rates thanks to few app giants which push the rates up. Maybe we could get better picture when we remove these extremly popular apps for each genre, and then check average again, but for now we´ll leave it.

### Popular apps by genre in Google Play
Data from Google Play market contains data about the numbers of installs, so it should be easier to get a view about genre popularity. 
But as we can see below, install numbers are not unfortunatelly precise enough, because there are lot of values which are open-ended(100+, 1000+, 5000+, etc).

In [37]:
display_table(final_android, 5)

1,000,000+ : 15.73 %
100,000+ : 11.55 %
10,000,000+ : 10.55 %
10,000+ : 10.2 %
1,000+ : 8.39 %
100+ : 6.92 %
5,000,000+ : 6.83 %
500,000+ : 5.56 %
50,000+ : 4.77 %
5,000+ : 4.51 %
10+ : 3.54 %
500+ : 3.25 %
50,000,000+ : 2.3 %
100,000,000+ : 2.13 %
50+ : 1.92 %
5+ : 0.79 %
1+ : 0.51 %
500,000,000+ : 0.27 %
1,000,000,000+ : 0.23 %
0+ : 0.05 %
0 : 0.01 %


Data are not clearly precise, because there is not clear wheather app with 100000+ installs has exactly 100000 installs or 200000 or even 350000 installs. Fortunately for our purpose we dont need very precise data, because we just want to know which genre attracts the most users.

For calculation there is a need to convert install numbers to float - to do it we need to remove all commas and plus characters, otherwise an error will appear.  

In [39]:
categories_android = freq_table(final_android, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in final_android:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', round(avg_n_installs, 2))

ART_AND_DESIGN : 1986335.09
AUTO_AND_VEHICLES : 647317.82
BEAUTY : 513151.89
BOOKS_AND_REFERENCE : 8767811.89
BUSINESS : 1712290.15
COMICS : 817657.27
COMMUNICATION : 38456119.17
DATING : 854028.83
EDUCATION : 1833495.15
ENTERTAINMENT : 11640705.88
EVENTS : 253542.22
FINANCE : 1387692.48
FOOD_AND_DRINK : 1924897.74
HEALTH_AND_FITNESS : 4188821.99
HOUSE_AND_HOME : 1331540.56
LIBRARIES_AND_DEMO : 638503.73
LIFESTYLE : 1437816.27
GAME : 15588015.6
FAMILY : 3695641.82
MEDICAL : 120550.62
SOCIAL : 23253652.13
SHOPPING : 7036877.31
PHOTOGRAPHY : 17840110.4
SPORTS : 3638640.14
TRAVEL_AND_LOCAL : 13984077.71
TOOLS : 10801391.3
PERSONALIZATION : 5201482.61
PRODUCTIVITY : 16787331.34
PARENTING : 542603.62
WEATHER : 5074486.2
VIDEO_PLAYERS : 24727872.45
NEWS_AND_MAGAZINES : 9549178.47
MAPS_AND_NAVIGATION : 4056941.77


Communications, Social and Photography are three top genres which attracts the most. For example in Communications those high numbers are pushed up again by few apps giants that have over five hundred milion installs or even over bilion installs..

In [40]:
for app in final_android:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                 or app[5] == '500,000,000+'
                                 or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Let´s look on genre Books and Reference that looks also pretty popular with an average number of installs of 8,767,811. 
It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Below some of the apps from this genre and number of installs:

In [45]:
for app in final_android:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

This genre include some variety of apps: for instance apps for reading ebooks, dictionaries, various collection of libraries, etc. But still we have here some kinds of apps that are pushing up the numbers...

In [46]:
for app in final_android:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                       or app[5] == '500,000,000+'
                                       or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Although there is only few such very popular apps, it looks that this genre still show a potential. Now we try to have a look on the apps, which are somewhere in the middle of popularity in terms of downloads [between 1,000,000 and 100,000,000].

In [47]:
for app in final_android:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                       or app[5] == '5,000,000+'
                                       or app[5] == '10,000,000+'
                                       or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

There are also quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

### Conclusion

This project analyzed data about the App Store and Google Play Store, with the goal of recommend an app profile that can be profitable for both markets.

Conclusion is that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. 